example/matrixMulWithMdspan/src/matrixMulMdSpan.cpp

```diff
@@ -0,0 +1,178 @@
+    using Dim = alpaka::DimInt<2>;
+
+    // Define matrix dimensions, A is MxK and B is KxN
+    const Idx M = 1024;
+    const Idx N = 512;
+    const Idx K = 1024;
+
+    // Define device and queue
+    using Acc = alpaka::AccCpuSerial<Dim, Idx>;
+    using Queue = alpaka::Queue<Acc, alpaka::Blocking>;
+    using Vec = alpaka::Vec<Dim, Idx>;
+
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+    auto const platformAcc = alpaka::Platform<Acc>{};
+    auto const devAcc = alpaka::getDevByIdx(platformAcc, 0);
+
+    Queue queue(devAcc);
+
+    // Define the 2D extents (dimensions)
+    Vec const extentA(static_cast<Idx>(M), static_cast<Idx>(K));
+    Vec const extentB(static_cast<Idx>(K), static_cast<Idx>(N));
+    Vec const extentC(static_cast<Idx>(M), static_cast<Idx>(N));
+
+    // Allocate host memory
+    auto bufHostA = alpaka::allocBuf<DataType, Idx>(devHost, extentA);
+    auto bufHostB = alpaka::allocBuf<DataType, Idx>(devHost, extentB);
+    auto bufHostC = alpaka::allocBuf<DataType, Idx>(devHost, extentC);
+
+    // Create mdspan view for bufHostA and bufHostB using alpaka::experimental::getMdSpan to fill the host buffers
+    auto mdHostA = alpaka::experimental::getMdSpan(bufHostA);
+    auto mdHostB = alpaka::experimental::getMdSpan(bufHostB);
+
+    // Initialize host matrices
+    for(Idx i = 0; i < M; ++i)
+    {
+        for(Idx j = 0; j < K; ++j)
+        {
+            // fill with some data
+            mdHostA(i, j) = static_cast<DataType>(i * K + j);
+        }
+    }
+    for(Idx i = 0; i < K; ++i)
+    {
+        for(Idx j = 0; j < N; ++j)
+        {
+            // fill with some data
+            mdHostB(i, j) = static_cast<DataType>(i * N + j);
+        }
+    }
+
+    // Allocate device memory
+    auto bufDevA = alpaka::allocBuf<DataType, Idx>(devAcc, extentA);
+    auto bufDevB = alpaka::allocBuf<DataType, Idx>(devAcc, extentB);
+    auto bufDevC = alpaka::allocBuf<DataType, Idx>(devAcc, extentC);
+
+    // Copy data to device, use directly host buffers (not mdspans used to fill the data)
+    alpaka::memcpy(queue, bufDevA, bufHostA);
+    alpaka::memcpy(queue, bufDevB, bufHostB);
+
+    // Create mdspan views for device buffers using alpaka::experimental::getMdSpan
+    auto mdDevA = alpaka::experimental::getMdSpan(bufDevA);
+    auto mdDevB = alpaka::experimental::getMdSpan(bufDevB);
+    auto mdDevC = alpaka::experimental::getMdSpan(bufDevC);
+
+    //  Let alpaka calculate good block and grid sizes given our full problem extent.
+    auto const workDiv = alpaka::getValidWorkDiv<Acc>(
+        devAcc,
+        extentC,
+        Vec::ones(),
+        false,
+        alpaka::GridBlockExtentSubDivRestrictions::Unrestricted);
+
+    // Execute the kernel
+    alpaka::exec<Acc>(queue, workDiv, MatrixMulKernel{}, mdDevA, mdDevB, mdDevC, K);
+
+    // Wait for the kernel to finish
+    alpaka::wait(queue);
+
```

Comment:
Not needed.

Reply:
I will keep this and added a comment saying "if the queue is blocking this is not needed."

Reply:
No, this is never needed, also is the queue is _not blocking_.

Reply:
The `wait` below (after the `memcpy`) is indeed needed, and it should always be there.

Reply:
I meant the last `wait` (after the `memcpy`), if the queue is blocking the host is blocked by `memcpy`  hence no need to `wait` after `memcpy`?  (and the queue is blocking-queue `using Queue = alpaka::Queue<Acc, alpaka::Blocking>;`)

Reply:
Yes.

If a queue is non-blocking, it is necessary to do `alpaka::wait(queue)` after the `memcpy` and before accessing the data on the host.

If a queue is blocking, it is not necessary to do `alpaka::wait(queue)`.
However, calling `alpaka::wait(queue)` should be cheap, though not completely free.
So maybe we can keep it anyway ?

Reply:
ok, I used `wait` only after `memcpy` statements and added no comments... Although the `queue` is blocking, as you said it is not expensive.